Assets/GameAssets/Player/PlayerScripts/Harpoon/HarpoonProjectileMovement.cs

```diff
@@ -66,4 +67,16 @@ private void HarpoonFiredProjectileMovement(Vector3 movement)
     {
         transform.position += movement;
     }
+
+    /// <summary>
+    /// Used to destroy the harpoon projectile when it hits something in scene.
+    /// </summary>
+    /// <param name="block"></param> what it collides against. 
+    private void OnTriggerEnter(Collider block)
+    {
+        if(block.gameObject.tag != "enemy")
```

Comment:
`if(!block.TryGetComponent<WeakPoint>(out Weakpoint weakpoint))` instead is much less likely to break or cause errors than a tag